Add constructors for default push rules

elegaanz opened the pull request:
Fixes #40

For the documentation, I just copied what was in the spec, I hope it is OK.

elegaanz committed:
Add constructors for default push rules

Fixes #40

jplatte commented on ruma-common/src/push.rs:
```diff
@@ -8,6 +8,7 @@ use serde::{Deserialize, Serialize};
 mod action;
 mod any_push_rule;
 mod condition;
+mod predefined_push_rules;
```
That's quite the long module name, I think `predefined` would suffice (or maybe `server_defaults` or sth like that).

jplatte commented on ruma-common/src/push/predefined_push_rules.rs:
```diff
@@ -0,0 +1,251 @@
+    /// # Parameters
+    ///
+    /// - `user_id`: the user for which to generate the default rules. Some rules depend on the
+    ///   user's ID (for instance those to send notifications when they are mentionned).
```
```suggestion
    ///   user's ID (for instance those to send notifications when they are mentioned).
```

jplatte commented on ruma-common/src/push/predefined_push_rules.rs:
```diff
@@ -0,0 +1,251 @@
+            actions: vec![Notify, SetTweak(Tweak::Highlight(false))],
+        }
+    }
+    /// Matches all encrypted events. Unlike other push rules,
```
There should be a newline before this doc comment. Also it seems to be linewrapped way before 100 chars.

elegaanz replied:
I wrapped them manually, my editor is not configured to wrap them after 100 chars I think. If it is important I can re-wrap them.

jplatte replied:
No, not that important. There's bound to be inconsistency anyway until we set up rustfmt to do it (just checked actually, maybe this already works with just a config option).

jplatte reviewed (commented):
Thanks for the PR! Looks good overall, just some nits.

elegaanz committed:
Fix small problems

- rename predefined_push_rules to predefined
- Fix a typo in a doc comment
- Add a blank line before doc comment
